src/main/java/mappers/firstvoices/CategoryMapper.java

```diff
@@ -3,63 +3,91 @@
  */
 package mappers.firstvoices;
 
-import mappers.propertyreaders.PropertyReader;
-import org.nuxeo.client.objects.Document;
-
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import mappers.propertyreaders.PropertyReader;
+import org.nuxeo.client.objects.Document;
+import org.nuxeo.client.objects.Documents;
 
 /**
  * @author dyona
  *
  */
-public class CategoryMapper extends RelatedPhraseMapper {
-	
-    @Override
-    protected String getCacheQuery() {
-    	// Include all categories in the Shared Data folder
-        return "SELECT * FROM FVCategory WHERE (fva:dialect IS NULL OR ecm:parentId='" + documents.get("Categories").getId()
-        		+ "') AND ecm:isTrashed = 0 AND ecm:isVersion = 0";
-    }
+public class CategoryMapper extends DictionaryCachedMapper {
 
-    private void updateMainDocumentReference(ArrayList<String> docIDList) {
-        documents.get("current").setPropertyValue(Properties.WORD_CATEGORIES, docIDList);
-    }
+  @Override
+  protected String getCacheQuery() {
 
-    public CategoryMapper() {
-        super("FVCategory", Columns.CATEGORIES);
-        parentKey = "Categories";
-        cacheProperty = Properties.TITLE;
-        propertyReaders.add(new PropertyReader(Properties.WORD_CATEGORIES, Columns.CATEGORIES));
+    Boolean useLocal = isLocalCategories();
+
+    /* For Local Dialect categories add running parameter localCategories */
+    Documents dialectCategoriesDirectory = client.operation("Repository.Query")
+        .param("query", "SELECT * FROM FVCategories "
+            + "WHERE fva:dialect = '" + getDialectID() + "' "
+            + "AND ecm:path STARTSWITH '/FV/Workspaces/' "
+            + "AND dc:title = 'Categories' "
+            + "AND ecm:isTrashed = 0 "
+            + "AND ecm:isVersion = 0")
+        .execute();
+    List<Document> documentsList = dialectCategoriesDirectory.streamEntries().collect(
+        Collectors.toList());
+
+    if (documentsList.size() > 0 && useLocal) {
+      Document categoriesDirectory = documentsList.get(0);
+      String categoriesDirectoryId = categoriesDirectory.getId();
+      return "SELECT * FROM FVCategory WHERE ecm:ancestorId = '" + categoriesDirectoryId
+          + "'AND ecm:isTrashed = 0 AND ecm:isVersion = 0";
     }
 
-    @Override
-    protected Document createDocument(Document doc, Integer depth) throws IOException {
+    /* Else return query for shared categories */
+    return "SELECT * FROM FVCategory WHERE fva:dialect IS NULL AND ecm:isTrashed = 0 AND ecm:isVersion = 0";
```

Comment:
This definitely needs  'ecm:isProxy =0' here (RE: FW-1445). Ideally, we would pass the Shared Categories directory that is cached in AbstractMigrator to this mapper - so that we can query based on ancestorId here too.

See: https://github.com/First-Peoples-Cultural-Council/fv-batch-import/blob/a710b192e133f0f8bdd18159d0df762a72478880/src/main/java/task/AbstractMigrator.java#L268

Then set here: https://github.com/First-Peoples-Cultural-Council/fv-batch-import/blob/a710b192e133f0f8bdd18159d0df762a72478880/src/main/java/task/FVWordMigrator.java#L37